.github/workflows/push.yaml

```diff
@@ -20,13 +20,29 @@ jobs:
         python-version: ["3.10", "3.11", "3.12"]
     steps:
       - uses: actions/checkout@v4
-      - name: Setup Python.
+      - name: Setup Python
         uses: actions/setup-python@v5
         with:
           python-version: ${{matrix.python-version}}
-      - name: Install dependencies.
+      - name: Install uv
         run: |
-          pip install poetry
-          poetry install --with dev
-      - name: Run tests.
-        run: poetry run pytest tests
+          pip install uv
+      - name: Install dependencies
+        shell: bash
+        run: |
+          uv venv
+          if [[ "$RUNNER_OS" == "Windows" ]]; then
+            source .venv/Scripts/activate
+          else
+            source .venv/bin/activate
+          fi
+          uv pip install -e ".[dev]"
```

Comment:
You shouldn't need to uv pip install here, you can just uv run pytest directly and have uv figure out dependencies from the pyproject.toml.

Reply:
Resolved. New version:

```
      - name: Install dependencies
        shell: bash
        run: |
          uv venv
          if [[ "$RUNNER_OS" == "Windows" ]]; then
            source .venv/Scripts/activate
          else
            source .venv/bin/activate
          fi
          uv run pytest
```

Reply:
The auto-tests actually gave me errors if I deleted this.

Reply:
I'm guessing you mean the complaints about a missing pytest? It's because you're not specifying that you're using the dev dependency group which you should be doing like this: `uv run --dev pytest tests`.

You also don't need to run `uv venv` above since `uv run` will automatically take care of that.